llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2545,7 +2545,8 @@ bool LoopAccessInfo::analyzeLoop(AAResults *AA, const LoopInfo *LI,
   for (StoreInst *ST : Stores) {
     Value *Ptr = ST->getPointerOperand();
 
-    if (isInvariant(Ptr)) {
+    const SCEV *PtrScev = replaceSymbolicStrideSCEV(*PSE, SymbolicStrides, Ptr);
```

Comment:
At first I couldn't really understand how this change is linked to the test changes in llvm/test/Transforms/LoopVectorize/version-stride-with-integer-casts.ll because there are no loop invariant pointers in that test. However, I guess the intention here is to simply add predicates for strides of 1 in the same way we do for loads below. And since you have a SCEV object it makes sense to call `PSE->getSE()->isLoopInvariant()` instead.

Reply:
The commit message says `use the information to refine the loop-invariant check`, but I can't think of any scenario where requiring a stride of 1 for the pointer would result in a different answer from `isLoopInvariant`. Prior to your change, when using a store pointer with a symbolic stride `isInvariant` would surely return false? With this patch applied the answer will still be false because now we know the stride is 1.

Reply:
Sorry, I got confused when writing the LAA test: I've updated the commit message now.